commit e98ed967f1662f002126cf8e0c996f2255578995 (refs/changes/90/390/6)
Author: Pedro P. Candel <pedro@joyent.com>
Date:   2016-09-06T18:11:55+02:00 (3 years, 1 month ago)
    
    TOOLS-1538 sdcadm should not throw ENOENT when checking for sdc-usbkey
    Reviewed by: David Pacheco <dap@joyent.com>
    Reviewed by: Trent Mick <trent.mick@joyent.com>

diff --git a/lib/common.js b/lib/common.js
index 4a5bad8..5f48816 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -1296,12 +1296,17 @@ function isUsbKeyMounted(log, cb) {
     function sdcUsbkeyExists(ctx, next) {
         fs.stat(ctx.utility, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.utilityExists = true;
             }
-            return next();
+            next();
         });
     }
 
@@ -1331,11 +1336,17 @@ function isUsbKeyMounted(log, cb) {
 
     function checkMeyMounted(ctx, next) {
         if (ctx.keyIsMounted || ctx.utilityExists) {
-            return next();
+            next();
+            return;
         }
         fs.stat(ctx.key, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.keyIsMounted = true;
@@ -1379,11 +1390,17 @@ function mountUsbKey(log, cb) {
 
     function sdcUsbkeyExists(ctx, next) {
         if (ctx.keyIsMounted) {
-            return next();
+            next();
+            return;
         }
         fs.stat(ctx.utility, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.utilityExists = true;
@@ -1484,16 +1501,23 @@ function unmountUsbKey(log, cb) {
 
     function sdcUsbkeyExists(ctx, next) {
         if (!ctx.keyIsMounted) {
-            return next();
+            next();
+            return;
         }
         fs.stat(ctx.utility, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.utilityExists = true;
             }
-            return next();
+            next();
+            return;
         });
     }
 
